modules/controller/dv-controller.js

```diff
@@ -164,125 +164,88 @@ class DVController {
-    //     });
-    //
-    //
-    //     await Models.handler_ids.update({
-    //         status: 'COMPLETED',
-    //     }, { where: { handler_id } });
-    // }
-
-    // async sendNetworkPurchase(dataSetId, erc725Identity, nodeId, otJsonObjectId, handlerId) {
-    //     const message = {
-    //         data_set_id: dataSetId,
-    //         dv_erc725_identity: erc725Identity,
-    //         handler_id: handlerId,
-    //         ot_json_object_id: otJsonObjectId,
-    //         wallet: this.config.node_wallet,
-    //     };
-    //     const dataPurchaseRequestObject = {
-    //         message,
-    //         messageSignature: Utilities.generateRsvSignature(
-    //             JSON.stringify(message),
-    //             this.web3,
-    //             this.config.node_private_key,
-    //         ),
-    //     };
-    //     await this.transport.sendDataPurchaseRequest(
-    //         dataPurchaseRequestObject,
-    //         nodeId,
-    //     );
-    // }
+    /**
+     * Handles permissioned data read request
+     * @param data_set_id - Dataset that holds permissioned data
+     * @param ot_object_id - Object id that holds permissioned data
+     * @param seller_node_id - Node id that holds permissioned data
+     * @param response - API result object
+     * @returns null
+     */
+    async handlePermissionedDataReadRequest(data_set_id, ot_object_id, seller_node_id, response) {
+        const handler_data = {
+            data_set_id,
+            ot_object_id,
+            seller_node_id,
+        };
+        const inserted_object = await Models.handler_ids.create({
+            status: 'PENDING',
+            data: JSON.stringify(handler_data),
+        });
+        const { handler_id } = inserted_object.dataValues;
+        this.logger.info(`Read private data with id ${ot_object_id} with handler id ${handler_id} initiated.`);
+
+        response.status(200);
+        response.send({
+            handler_id,
+        });
+
+        this.commandExecutor.add({
+            name: 'dvPermissionedDataReadRequestCommand',
+            delay: 0,
+            data: {
+                data_set_id,
+                ot_object_id,
+                seller_node_id,
+                handler_id,
+            },
+            transactional: false,
+        });
+    }
+
+    async handlePermissionedDataReadResponse(message) {
+        const {
+            handler_id, ot_objects,
+        } = message;
+        const documentsToBeUpdated = [];
+        ot_objects.forEach((otObject) => {
+            otObject.relatedObjects.forEach((relatedObject) => {
+                if (relatedObject.vertex.vertexType === 'Data') {
+                    if (ImportUtilities.calculatePermissionedDataHash(relatedObject.vertex
+                        .data.permissioned_data) === relatedObject.vertex.data.permissioned_data
+                        .permissioned_data_hash) {
+                        documentsToBeUpdated.push(relatedObject.vertex);
```

Comment:
Permissioned data hash not matching with the hash calculated upon receiving data is alarming, we should throw some kind of error here, not just ignore that ot_object

Reply:
We'll add handling of that error when we implement purchase disputes, but it's good to have the error thrown somewhere instead of searching for it later.

Reply:
I agree